db-queue-spring/src/main/java/ru/yoomoney/tech/dbqueue/spring/dao/H2QueuePickTaskDao.java

```diff
@@ -0,0 +1,288 @@
+package ru.yoomoney.tech.dbqueue.spring.dao;
+
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.transaction.support.TransactionOperations;
+import ru.yoomoney.tech.dbqueue.api.TaskRecord;
+import ru.yoomoney.tech.dbqueue.config.QueueTableSchema;
+import ru.yoomoney.tech.dbqueue.dao.PickTaskSettings;
+import ru.yoomoney.tech.dbqueue.dao.QueuePickTaskDao;
+import ru.yoomoney.tech.dbqueue.settings.QueueLocation;
+import ru.yoomoney.tech.dbqueue.settings.TaskRetryType;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.sql.*;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class H2QueuePickTaskDao implements QueuePickTaskDao {
+    private final JdbcOperations jdbcTemplate;
+    private final QueueTableSchema queueTableSchema;
+    private final PickTaskSettings pickTaskSettings;
+
+    public H2QueuePickTaskDao(@Nonnull final JdbcOperations jdbcOperations,
+                              @Nonnull final QueueTableSchema queueTableSchema,
+                              @Nonnull final PickTaskSettings pickTaskSettings,
+                              @Nonnull final TransactionOperations transactionTemplate) {
+
+        this.jdbcTemplate = Objects.requireNonNull(jdbcOperations, "jdbc template can't be null");
+        this.queueTableSchema = Objects.requireNonNull(queueTableSchema, "table schema can't be null");
+        this.pickTaskSettings = Objects.requireNonNull(pickTaskSettings, "settings can't be null");
+        Objects.requireNonNull(transactionTemplate, "transaction template must be not null");
+
+        transactionTemplate.executeWithoutResult(transactionStatus -> {
+            jdbcTemplate.update(
+                    String.format(
+                            "CREATE ALIAS IF NOT EXISTS PICK_TASK FOR \"%s.pickTask\"",
+                            PickerProcedure.class.getName()));
+        });
+    }
+
+    @Nullable
+    @Override
+    public TaskRecord pickTask(@Nonnull final QueueLocation location) {
+        Objects.requireNonNull(location, "location can't be null");
+
+        return jdbcTemplate.query(
+                PickerProcedure.createPickQuery(location, pickTaskSettings, queueTableSchema),
```

Comment:
Why do you choose the procedure? You can simply do 3 consecutive calls to spring jdbc template. Is there any difference?

Reply:
There is no need to use it at all. I agree with you, it would be better to use pure java methods.